qiskit/optimization/algorithms/recursive_minimum_eigen_optimizer.py

```diff
@@ -129,7 +129,9 @@ def solve(self, problem: OptimizationProblem) -> OptimizationResult:
             x_j = problem_.variables.get_names(j)
             if correlations[i, j] > 0:
                 # set x_i = x_j
-                problem_ = problem_.substitute_variables(variables=SparseTriple([i], [j], [1]))
+                problem_, status = problem_.substitute_variables(
+                    variables=SparseTriple([i], [j], [1]))
+                assert status == problem_.substitute_status.ok
```

Comment:
Even if an substitution is valid, the resulting problem may be infeasible. I think it would be more informative to tell the status to users. I discussed it with @stefan-woerner. He agreed my idea.

Reply:
I mean the error is raised anyways if the assertion fails, but it would say "Assertion failed: status == problem_.substitute_status.ok". While if we put a helpful message, the user will know more quickly what went wrong 🙂

Reply:
I agree. Done.